### ISSUE TITLE
Danger crashes on travis ci with error from gitlab

### ISSUE THREAD

**Issue by bjankord** [danger/danger-js#884]
We recently updated to danger v9.0.1 and it was working just fine. However we recently noticed the following error showing up on travis when we run danger.

```
Log:
Creating network "terracore_default" with the default driver
Creating terracore_standalone-chrome_1 ... 
Creating terracore_standalone-chrome_1
> terra-core@0.1.0 danger /opt/module
> danger ci
/opt/module/node_modules/gitlab/dist/index.js:1
(function (exports, require, module, __filename, __dirname) { "use strict";function _interopDefault(e){return e&&"object"==typeof e&&"default"in e?e.default:e}Object.defineProperty(exports,"__esModule",{value:!0});var universalUrl=require("universal-url"),Li=_interopDefault(require("li")),humps=require("humps"),Ky=_interopDefault(require("ky-universal")),FormData=_interopDefault(require("form-data")),queryString=require("query-string"),randomstring=_interopDefault(require("randomstring")),extendStatics=function(e,t){return(extendStatics=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(e,t){e.__proto__=t}||function(e,t){for(var r in t)t.hasOwnProperty(r)&&(e[r]=t[r])})(e,t)};function __extends(e,t){function r(){this.constructor=e}extendStatics(e,t),e.prototype=null===t?Object.create(t):(r.prototype=t.prototype,new r)}var __assign=function(){return(__assign=Object.assign||function(e){for(var t,r=1,n=arguments.length;r<n;r++)for(var o in t=argument
TypeError: global$1.XMLHttpRequest is not a constructor
    at checkTypeSupport (/opt/module/node_modules/gitlab/dist/index.js:1:32017)
    at Object.<anonymous> (/opt/module/node_modules/gitlab/dist/index.js:1:32341)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/opt/module/node_modules/danger/distribution/platforms/gitlab/GitLabAPI.js:39:16)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! terra-core@0.1.0 danger: `danger ci`
npm ERR! Exit status 1
npm ERR! 
```

It looks like there may be an issue in the [gitlab package](https://github.com/jdalrymple/node-gitlab) as this line alludes to:
```
TypeError: global$1.XMLHttpRequest is not a constructor
    at checkTypeSupport (/opt/module/node_modules/gitlab/dist/index.js:1:32017)
```
It looks like [a new 6.x minor version of gitlab was just released a few hours ago](https://github.com/jdalrymple/node-gitlab/releases/tag/6.4.0) that may be triggering this breaking change.

**Comment by darcy-rayner** [comment]
I came here to comment on this as well. Just started seeing an identical error.

**Comment by f-meloni** [comment]
`danger-swift` and `danger-kotlin` PRs are failing for the same reason

```
/Users/travis/.nvm/versions/node/v10.15.0/lib/node_modules/danger/node_modules/gitlab/dist/index.js:1
(function (exports, require, module, __filename, __dirname) { "use strict";function _interopDefault(e){return e&&"object"==typeof e&&"default"in e?e.default:e}Object.defineProperty(exports,"__esModule",{value:!0});var universalUrl=require("universal-url"),Li=_interopDefault(require("li")),humps=require("humps"),Ky=_interopDefault(require("ky-universal")),FormData=_interopDefault(require("form-data")),queryString=require("query-string"),randomstring=_interopDefault(require("randomstring")),extendStatics=function(e,t){return(extendStatics=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(e,t){e.__proto__=t}||function(e,t){for(var r in t)t.hasOwnProperty(r)&&(e[r]=t[r])})(e,t)};function __extends(e,t){function r(){this.constructor=e}extendStatics(e,t),e.prototype=null===t?Object.create(t):(r.prototype=t.prototype,new r)}var __assign=function(){return(__assign=Object.assign||function(e){for(var t,
TypeError: global$1.XMLHttpRequest is not a constructor
    at checkTypeSupport (/Users/travis/.nvm/versions/node/v10.15.0/lib/node_modules/danger/node_modules/gitlab/dist/index.js:1:32017)
    at Object.<anonymous> (/Users/travis/.nvm/versions/node/v10.15.0/lib/node_modules/danger/node_modules/gitlab/dist/index.js:1:32341)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/Users/travis/.nvm/versions/node/v10.15.0/lib/node_modules/danger/distribution/platforms/gitlab/GitLabAPI.js:39:16)
```

**Comment by tyankatsu0105** [comment]
👀
https://github.com/jdalrymple/node-gitlab/issues/377

**Comment by IljaDaderko** [comment]
Doesn't seem to be specific to travis or gitlab, seeing same error when running within GitHub actions as well.

**Comment by f-meloni** [comment]
I've opened https://github.com/danger/danger-js/pull/885, I hope that fixes the problem

**Comment by ashfurrow** [comment]
I'm seeing this too, also on Danger-Swift ([example failure](https://travis-ci.org/artsy/eigen/builds/552776451)). I think this is happening since a new version of `node-gitlab` was introduced [with the bug](https://github.com/jdalrymple/node-gitlab/issues/377) that @tyankatsu0105 linked to, and since we installed Danger-Swift with `npm install -g danger`, we get the latest version of its dependencies (no lock file, so no pinning).

I think the "right" solution is just to wait for the GitLab bug to get fixed, then things should "just work" for Danger-Swift. If you need a workaround in the meantime, you can explore either adding a Yarn/NPM lockfile to pin to an older version of `node-gitlab`, or using NPM to install a specific version of `node-gitlab` _before_ installing Danger (so that the older version is used as a dependency). 

```sh
npm install -g gitlab@6.3.7
npm install -g danger
```

Which _should_ work, though I haven't yet tried it.

**Comment by ashfurrow** [comment]
Seeing the fix from #885 work as expected: https://github.com/artsy/eigen/pull/2854 Thanks @orta and @f-meloni!
